### ISSUE TITLE
Why I get different ADB2C refresh token lifetime for apps in the same tennant?

### ISSUE THREAD

**Issue by miranc** [azure-ad-b2c/samples#527]
I have two react SPA apps. Both use the same msal-react and msal-browser package versions for authentication.
Both use the same sign-in flow and therefore have the same token configuration.
One is deployed as an app service in Azure and the second one is deployed as an app on virtual machine IIS in Azure. App in app service gets refresh token that lasts 14 days (as configured in sign-in flow) and app deployed in Azure virtual machine gets refresh token valid only 24h. Authentication flow is the same. Each app has its own app registration, scopes and api resources are different but configured correctly. Besides custom scopes they both have Graph Api offline_access and openid scopes.
What could be the reason that app published to IIS in virtual machine in Azure gets only 24h refresh token? Redirect URIs in app registration and authentication settings are different for both apps and set to return to the app itself. Both authentications work just that refresh token lifetimes are different, and that causes often login redirects for an app deployed to virtual machine IIS.
I have tried many suggested things like adding domain and login hints to silentLoginRequest and loginRedirectRequest but I always get kicked to login page after 24h.

**Comment by JasSuri** [comment]
If redirect URI is set for App Type: SPA, on the app registration, then the refresh token will always be 24h.

https://learn.microsoft.com/en-us/azure/active-directory-b2c/configure-tokens?pivots=b2c-custom-policy#token-lifetime-behavior

**Comment by miranc** [comment]
@JasSuri Yes, redirect URI is set to SAP, but it is the case in both apps. Both are SPA, and I'm aware that SPAs have a limitation of 24h refresh-token, but somehow second app is not detected as SPA and gets 14, 60 days token. And it is spa, app made in react, index.htnl contains only <div id="root", network flow while using app is continuous, no page reloads while using app... So I'm interested how is this possible, and how I can replicate behavior on both apps? Because sending user to login page every 24h is really bad UX in my opinion.
